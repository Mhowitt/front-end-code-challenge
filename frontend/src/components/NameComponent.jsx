import React from 'react';
import { Link } from 'react-router-dom';

const NameComponent = props => {
  return (
    <div>
      <h1>Let's Learn a bit more about you!</h1>
      <label htmlFor="firstName">First Name: </label>
      <input type="text" name="firstName" onChange={props.handleChange} value={props.firstName}/>
      <label htmlFor="lastName">last Name: </label>
      <input type="text" name="lastName" onChange={props.handleChange} value={props.lastName}/>
      <button><Link to="/profile/gender"> Next</Link></button>

    </div>
  )
}

export default NameComponent;