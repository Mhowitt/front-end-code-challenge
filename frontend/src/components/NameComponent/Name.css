.login_field {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin-top: 15%;
  align-items: center;
  position: relative;
}

.login_field > .login_input {
  border-bottom: 1px solid rgb(213, 189, 187);
  box-shadow: rgb(223, 212, 211) 0px 6px 15px -5px;
  margin-right: 0;
  bottom: 2px;
  left: 3px;
  position: relative;
  border-top: none;
  border-right: none;
  border-left: none;
  width: 50%;
  height: 40px;
}

.login_field > .login_input::placeholder {
  color: #9099a2;
  text-transform: uppercase;
  text-align: center;
}

.login_input:focus {
  outline: none;
  box-shadow: rgb(223, 212, 211) 0px 6px 15px -5px;
  text-align: center;
}

.name-button {
  width: auto;
  height: 40px;
  border-radius: 10px;
  background-color: #96858f;
  color: rgb(254, 252, 252);
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
}

.button-container {
  margin-top: 2%;
}
