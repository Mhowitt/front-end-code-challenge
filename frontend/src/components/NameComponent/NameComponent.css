.login_field {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin-top: 15%;
  align-items: center;
  position: relative;

}

.login_field > .login_input {
  border-bottom: 1px solid rgb(213,189,187);
  box-shadow: rgb(223, 212, 211) 0px 6px 15px -5px;
  margin-right: 0;
  bottom: 2px;
  left: 3px;
  position: relative;
  border-top: none;
  border-right: none;
  border-left: none;
  width: 50%;
  height: 40px;
}

.login_field > .login_input::placeholder {
  color: #9099A2;
  text-transform: uppercase;
  text-align: center;
}

.login_input:focus {
  outline: none;
  box-shadow: rgb(223, 212, 211) 0px 6px 15px -5px;
  text-align: center;
}