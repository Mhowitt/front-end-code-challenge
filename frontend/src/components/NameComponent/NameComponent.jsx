import React from 'react';
import { Link } from 'react-router-dom';
import'./NameComponent.css'
import ProgressBar from '../ProgressBar/ProgressBar.jsx'

const NameComponent = props => {
  return (
    <div>
      <h2>Let's Learn a bit more about you!</h2>
      <ProgressBar question={props.question}/>
      <div className="card-container" >
        <div className="login_field">
        <input type="text" name="firstName" className="login_input" onChange={props.handleChange} value={props.firstName} placeholder="What's your Name" />
        </div>
        <div className="next-button">
          <Link to="/profile/gender"><div className="arrow"></div></Link>
        </div>
      </div>
    </div>
  )
}

export default NameComponent;