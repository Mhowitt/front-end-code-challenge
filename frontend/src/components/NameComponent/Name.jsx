import React from 'react';
import { Link } from 'react-router-dom';
import './Name.css';

const Name = props => {
  return (
    <div>
      <h2>Let's learn a bit more about you!</h2>
      <div className="card-container">
        <div className="login_field">
          <input
            type="text"
            name="firstName"
            className="login_input"
            onChange={props.handleChange}
            value={props.firstName}
            placeholder="What's your Name"
          />
        </div>
        {props.firstName.length === 0 ? (
          ''
        ) : (
          <div className="button-container">
            <Link to="/profile/gender">
              <button onClick={props.handleClick} className="name-button">
                {' '}
                Let's get started{' '}
              </button>
            </Link>
          </div>
        )}
      </div>
    </div>
  );
};

export default Name;
