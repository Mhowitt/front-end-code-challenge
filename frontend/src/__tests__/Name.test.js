import React from 'react'; // for JSX
import { shallow, mount } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import NameComponent from '../components/NameComponent/Name.jsx';

describe('<NameComponent />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = shallow(<NameComponent firstName="test" />);
  });
  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
