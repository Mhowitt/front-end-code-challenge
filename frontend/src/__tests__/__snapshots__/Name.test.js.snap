// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NameComponent /> matches snapshot 1`] = `
<div>
  <h2>
    Let's learn a bit more about you!
  </h2>
  <div
    className="card-container"
  >
    <div
      className="login_field"
    >
      <input
        className="login_input"
        name="firstName"
        placeholder="What's your Name"
        type="text"
        value="test"
      />
    </div>
    <div
      className="button-container"
    >
      <Link
        replace={false}
        to="/profile/gender"
      >
        <button
          className="name-button"
        >
           
          Let's get started
           
        </button>
      </Link>
    </div>
  </div>
</div>
`;
